pkg/declextract/declextract.go

```diff
@@ -386,8 +408,13 @@ func (ctx *context) fieldTypeArray(f *Field, parent string) string {
 		Type: t.Elem,
 	}
 	elemType := ctx.fieldType(elem, nil, parent, true)
-	if t.MinSize == 1 && t.MaxSize == 1 {
-		return elemType
+	if t.IsConstSize {
+		switch t.MaxSize {
```

Comment:
Nit: it would be nice to have a comment on why MaxSize=0 and MaxSize=1 are special here.

Reply:
Done.